conda/cli/install.py

```
@@ -172,119 +170,201 @@ def get_revision(arg, json=False):
-        "channel_urls": context_channels,
+        "channel_urls": context.channels,
         # TODO: deprecate --unknown
         # "unknown": args.unknown,  # --unknown
         "prepend": not args.override_channels,  # --override-channels
         "use_local": args.use_local,  # --use-local
     }
 
+
+def ensure_update_specs_exist(prefix: str, specs: list[str]):
+    """Checks that each spec that is requested as an update exists in the prefix
+
+    :param prefix: The target install prefix
+    :param specs: List of specs to be updated
+    :raises CondaError: if there is an invalid spec provided
+    :raises PackageNotInstalledError: if the requested specs to install don't exist in the prefix
+    """
+    prefix_data = PrefixData(prefix)
+    for spec in specs:
+        spec = MatchSpec(spec)
+        if not spec.is_name_only_spec:
+            raise CondaError(
+                f"Invalid spec for 'conda update': {spec}\nUse 'conda install' instead."
+            )
+        if not prefix_data.get(spec.name, None):
+            raise PackageNotInstalledError(prefix, spec.name)
+
+
+def install(args, parser, command="install"):
+    """Logic for `conda install`, `conda update`, and `conda create`."""
+    prefix = context.target_prefix
+    newenv = bool(command == "create")
+    isupdate = bool(command == "update")
+    isinstall = bool(command == "install")
+    isremove = bool(command == "remove")
+
+    # common validations for all types of installs
+    validate_install_command(prefix=prefix, newenv=newenv)
+
+    # this is sort of a hack.  current_repodata.json may not have any .tar.bz2 files,
+    #    because it deduplicates records that exist as both formats.  Forcing this to
+    #    repodata.json ensures that .tar.bz2 files are available
+    if context.use_only_tar_bz2:
+        args.repodata_fns = ("repodata.json",)
+
+    # ensure trash is cleared from existing prefix
+    if isdir(prefix):
+        delete_trash(prefix)
```

Should we swallow exceptions here if there's an error while cleaning up? Maybe it shouldn't be fatal.